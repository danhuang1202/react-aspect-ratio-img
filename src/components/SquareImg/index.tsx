import React, { FunctionComponent, ReactNode } from 'react'
import { SquareFence } from 'react-aspect-ratio-fence'
import 'react-aspect-ratio-fence/css/components/SquareFence/style.css'

// @ts-ignore
import styles from './style.css'

type Props = {
  /**
   *  outter html tag name
   *  @default div
   * */
  outerElementType?: string
  /** Custom class name */
  customClass?: string
  /** The src attribute of <img /> tag */
  src: string
  /** Attributes of <img /> tag */
  imgAttributes?: object
  /** custom children node */
  children?: ReactNode
}

const SquareImg: FunctionComponent<Props> = ({
  outerElementType = 'div',
  customClass,
  src,
  imgAttributes,
  children
}) => {
  return (
    <SquareFence elementType={outerElementType} customClass={customClass}>
      <img alt="" {...imgAttributes} src={src} className={styles.img} />
      {children}
    </SquareFence>
  )
}

export default SquareImg
