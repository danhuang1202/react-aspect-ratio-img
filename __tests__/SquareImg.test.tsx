import React from 'react'
import renderer from 'react-test-renderer'
import SquareImg from '@components/SquareImg'

describe('[snapshot] SquareImg', () => {
  const defaultProps = {
    src: 'SquareImg.png',
    imgAttributes: {
      alt: 'SquareImg',
      srcSet: `
        square-320w.jpg,
        square-480w.jpg 1.5x,
        square-640w.jpg 2x`
    }
  }

  it('SquareImg', () => {
    const dom = renderer.create(<SquareImg {...defaultProps} />)
    expect(dom).toMatchSnapshot()
  })

  it('SquareImg with outerElementType', () => {
    const props = {
      ...defaultProps,
      outerElementType: 'span'
    }

    const dom = renderer.create(<SquareImg {...props} />)
    expect(dom).toMatchSnapshot()
  })
})
