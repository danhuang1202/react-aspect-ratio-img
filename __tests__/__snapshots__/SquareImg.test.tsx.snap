// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] SquareImg SquareImg 1`] = `
<div
  className="SquareFence_wrap_3jTo2"
>
  <img
    alt="SquareImg"
    className="SquareImg_img_2gAcq"
    src="SquareImg.png"
    srcSet="
        square-320w.jpg,
        square-480w.jpg 1.5x,
        square-640w.jpg 2x"
  />
</div>
`;

exports[`[snapshot] SquareImg SquareImg with outerElementType 1`] = `
<span
  className="SquareFence_wrap_3jTo2"
>
  <img
    alt="SquareImg"
    className="SquareImg_img_2gAcq"
    src="SquareImg.png"
    srcSet="
        square-320w.jpg,
        square-480w.jpg 1.5x,
        square-640w.jpg 2x"
  />
</span>
`;
